<launch>
    <arg name="map_size_x" value="30.0" />
    <arg name="map_size_y" value="30.0" />
    <arg name="map_size_z" value=" 5.0" />

    <arg name="odom_topic" value="visual_slam/odom" />

    <arg name="pcd_file" value="$(find px4_swarm)/../../assets/outdoor.pcd" />
    <node name="pointcloud_server" pkg="pcl_ros" type="pcd_to_pointcloud" args="$(arg pcd_file) 1.0" output="screen">
        <param name="frame_id" value="world" />
        <remap from="/cloud_pcd" to="/map_generator/global_cloud" />
    </node>

    <include file="$(find px4_swarm)/launch/run_in_sim.launch">
        <arg name="drone_id" value="0" />

        <arg name="init_x" value="0.0" />
        <arg name="init_y" value="-1.0" />
        <arg name="init_z" value="0.0" />

        <arg name="map_size_x" value="$(arg map_size_x)" />
        <arg name="map_size_y" value="$(arg map_size_y)" />
        <arg name="map_size_z" value="$(arg map_size_z)" />

        <arg name="odom_topic" value="$(arg odom_topic)" />
    </include>

    <include file="$(find px4_swarm)/launch/run_in_sim.launch">
        <arg name="drone_id" value="1" />

        <arg name="init_x" value="0.0" />
        <arg name="init_y" value="0.0" />
        <arg name="init_z" value="0.0" />

        <arg name="map_size_x" value="$(arg map_size_x)" />
        <arg name="map_size_y" value="$(arg map_size_y)" />
        <arg name="map_size_z" value="$(arg map_size_z)" />

        <arg name="odom_topic" value="$(arg odom_topic)" />
    </include>

    <include file="$(find px4_swarm)/launch/run_in_sim.launch">
        <arg name="drone_id" value="2" />

        <arg name="init_x" value="0.0" />
        <arg name="init_y" value="1.0" />
        <arg name="init_z" value="0.0" />

        <arg name="map_size_x" value="$(arg map_size_x)" />
        <arg name="map_size_y" value="$(arg map_size_y)" />
        <arg name="map_size_z" value="$(arg map_size_z)" />
        <arg name="odom_topic" value="$(arg odom_topic)" />
    </include>

    <include file="$(find px4_swarm)/launch/run_in_sim.launch">
        <arg name="drone_id" value="3" />

        <arg name="init_x" value="1.0" />
        <arg name="init_y" value="-1.0" />
        <arg name="init_z" value="0.0" />

        <arg name="map_size_x" value="$(arg map_size_x)" />
        <arg name="map_size_y" value="$(arg map_size_y)" />
        <arg name="map_size_z" value="$(arg map_size_z)" />
        <arg name="odom_topic" value="$(arg odom_topic)" />
    </include>

    <include file="$(find px4_swarm)/launch/run_in_sim.launch">
        <arg name="drone_id" value="4" />

        <arg name="init_x" value="1.0" />
        <arg name="init_y" value="0.0" />
        <arg name="init_z" value="0.0" />

        <arg name="map_size_x" value="$(arg map_size_x)" />
        <arg name="map_size_y" value="$(arg map_size_y)" />
        <arg name="map_size_z" value="$(arg map_size_z)" />
        <arg name="odom_topic" value="$(arg odom_topic)" />
    </include>

    <include file="$(find px4_swarm)/launch/run_in_sim.launch">
        <arg name="drone_id" value="5" />

        <arg name="init_x" value="1.0" />
        <arg name="init_y" value="1.0" />
        <arg name="init_z" value="0.0" />

        <arg name="map_size_x" value="$(arg map_size_x)" />
        <arg name="map_size_y" value="$(arg map_size_y)" />
        <arg name="map_size_z" value="$(arg map_size_z)" />
        <arg name="odom_topic" value="$(arg odom_topic)" />
    </include>

    <include file="$(find px4_swarm)/launch/run_in_sim.launch">
        <arg name="drone_id" value="6" />

        <arg name="init_x" value="2.0" />
        <arg name="init_y" value="-1.0" />
        <arg name="init_z" value="0.0" />

        <arg name="map_size_x" value="$(arg map_size_x)" />
        <arg name="map_size_y" value="$(arg map_size_y)" />
        <arg name="map_size_z" value="$(arg map_size_z)" />
        <arg name="odom_topic" value="$(arg odom_topic)" />
    </include>

    <include file="$(find px4_swarm)/launch/run_in_sim.launch">
        <arg name="drone_id" value="7" />

        <arg name="init_x" value="2.0" />
        <arg name="init_y" value="0.0" />
        <arg name="init_z" value="0.0" />

        <arg name="map_size_x" value="$(arg map_size_x)" />
        <arg name="map_size_y" value="$(arg map_size_y)" />
        <arg name="map_size_z" value="$(arg map_size_z)" />
        <arg name="odom_topic" value="$(arg odom_topic)" />
    </include>

    <include file="$(find px4_swarm)/launch/run_in_sim.launch">
        <arg name="drone_id" value="8" />

        <arg name="init_x" value="2.0" />
        <arg name="init_y" value="1.0" />
        <arg name="init_z" value="0.0" />

        <arg name="map_size_x" value="$(arg map_size_x)" />
        <arg name="map_size_y" value="$(arg map_size_y)" />
        <arg name="map_size_z" value="$(arg map_size_z)" />
        <arg name="odom_topic" value="$(arg odom_topic)" />
    </include>

    <include file="$(find px4_swarm)/launch/run_in_sim.launch">
        <arg name="drone_id" value="9" />

        <arg name="init_x" value="2.0" />
        <arg name="init_y" value="2.0" />
        <arg name="init_z" value="0.0" />

        <arg name="map_size_x" value="$(arg map_size_x)" />
        <arg name="map_size_y" value="$(arg map_size_y)" />
        <arg name="map_size_z" value="$(arg map_size_z)" />
        <arg name="odom_topic" value="$(arg odom_topic)" />
    </include>
</launch>